pkg/time/time.go

```diff
@@ -72,3 +66,58 @@ type (
 	Timer      = time.Timer
 	Weekday    = time.Weekday
 )
+
+var (
+	MaxInternalTimerDelay time.Duration
+)
+
+// After overrides the stdlib time.After to enforce maximum sleepiness via
+// option.MaxInternalTimerDelay.
+func After(d Duration) <-chan Time {
+	if MaxInternalTimerDelay > 0 && d > MaxInternalTimerDelay {
+		d = MaxInternalTimerDelay
```

Comment:
Will this only be used in testing, should we emit a log that something is having time.After/Sleep capped?

Reply:
No harm emitting a debug log here yeah. Could help folks understand better in future if they are unaware of this change.

Reply:
Added to https://github.com/cilium/cilium/issues/28844